src/lib.rs

```diff
@@ -143,19 +141,17 @@
 //! ### Compact type with HasCompact
 //!
 //! ```
-//! # // Import macros if derive feature is not used.
-//! # #[cfg(not(feature="derive"))]
-//! # use parity_scale_codec_derive::{Encode, Decode};
+//! # use parity_scale_codec_derive::{Encode as DeriveEncode, Decode as DeriveDecode};
 //!
 //! use parity_scale_codec::{Encode, Decode, Compact, HasCompact};
 //!
-//! #[derive(Debug, PartialEq, Encode, Decode)]
+//! #[derive(Debug, PartialEq, DeriveEncode, DeriveDecode)]
```

Comment:
I'm not happy with this. People will be confused on where `DeriveEncode` is coming from. 

https://users.rust-lang.org/t/how-to-document-optional-features-in-api-docs/64577/3 above the code example with `feature = "derive"` would be really nice!

Reply:
This auto-generated feature warning seems to only work on whole modules or types. Not on single doc lines, so i added:  
```rust
#![cfg_attr(feature = "derive", doc = "```rust")]
#![cfg_attr(not(feature = "derive"), doc = "*(Only compiled with feature `derive`)*\n```ignore")]
```
To each example to not compile them without the feature.